templates/user-data.sh.tmpl

```diff
@@ -7,7 +7,7 @@ exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
 # -----------------------------------------------------------------------------
 
 function install_docker_ce() {
-  sudo yum install -y docker-20.10.4-1.amzn2
+  sudo amazon-linux-extras install docker
```

Comment:
@colmsnowplow do you think its worth pinning to a particular version of Docker?  Worried that unpinned it might install a broken package / something incompatible again...

Reply:
hrmm... we are not using any of the latest features of docker in this setup so I think lowering recurrence of the issue feels better.  Lets go with option 2.